### DISCUSSION TITLE
How can a student submit an assignment?

### DISCUSSION THREAD

**Discussion by ItaiBEdu** [community/community#65792] [GitHub Education] [answered]
I can't figure out how a student can submit an assignment? I was able to create an assignment with my teacher account, invite my student account, accept the assignment by my student account, make some changes to the template repository, then commit the changes. However, when I try to push the changes I get the following error message:

> Updates were rejected because the tip of your current branch is behind
> hint: its remote counterpart. Integrate the remote changes (e.g.
> hint: 'git pull ...') before pushing again.

If I try to pull I get the following error message:

> You have divergent branches and need to specify how to reconcile them.
> hint: You can do so by running one of the following commands sometime before
> hint: your next pull:
> hint: 
> hint:   git config pull.rebase false  # merge
> hint:   git config pull.rebase true   # rebase
> hint:   git config pull.ff only       # fast-forward only

Why do I need to reconcile divergent branches? I haven't made any changes to the repository in my teacher account. Why can't I submit my assignment?

**Comment by seyLu** [comment]
#### Hi @ItaiBDalet 

#### To solve this issue:
- try to fast-forward pull

```bash
git pull --ff-only
```

- if that doesn't work, rebase pull
```bash
git pull --rebase
```

#### You usually encounter this type of errors when the remote history diverged from the local history -- which is normal when working on a team setting

**Reply by ItaiBEdu** [reply]
I did as you suggested. `git pull --ff-only` didn't work, but `git pull --rebase` did. I was then able to perform a `push`. Nevertheless, my teacher's assignment page doesn't indicate that the student has submitted the assignment. Why?

![NotSubmitted](https://github.com/community/community/assets/143313920/13427a64-11bc-4bb3-a559-6a0cff6f28a5)

**Reply by AbdelwahabAdam** [reply]
There are a few reasons why your teacher's assignment page might not indicate that the student has submitted the assignment.

The teacher may not have checked their GitHub notifications yet.
The teacher may have set up their GitHub repository to automatically merge pull requests from students, but this may not have happened yet.
There may be a delay in the GitHub system.
If you have been able to push your changes to the remote repository, then the assignment has been submitted successfully. You can try refreshing the teacher's assignment page to see if it has been updated. If you are still having problems, you can contact your teacher for help.

**Reply by ItaiBEdu** [reply]
I am both the teacher and the student, using different accounts. I pushed the changes more than 4 hours ago from the student account, but the teacher account still doesn't indicate that the assignment has been submitted. The same screenshot I embedded in my original post is valid at the current point of time.

**Comment by ItaiBEdu** [comment, marked as answer]
If a submission deadline was set for the assignment, the submissions are indicated on the teacher's side only once the deadline has been reached.

**Comment by ghost** [comment]
https://colab.research.google.com/drive/11ti4FSldGjKD7C0u4MEKd99mfLWx2dm1?usp=sharing
